dist/logs/logzioLogsFunction/core.js

```diff
@@ -1,60 +1,186 @@
+    }
+};
+const getCallBackFunction = context => {
+    return function callback(err, bulk) {
+        if (err) {
+            context.err(`logzio-logger error: ${err}`, err);
+        }
+        context.log('Done.');
+        context.done();
+    };
+};
+const getParserOptions = () => {
+    return {
+        token: process.env.LogzioLogsToken,
+        host: process.env.LogzioLogsHost,
+        bufferSize: Number(process.env.BufferSize),
+        timeout: Number(process.env.Timeout),
+    };
+};
+const getContainerDetails = () => {
+    return {
+        storageConnectionString: process.env.LogsStorageConnectionString,
+        containerName: containerName,
+    };
+};
+const deleteDirsWithFiles = (foldersToDelete, context) => {
+    foldersToDelete.forEach(folderPath => {
+        fs.rmdirSync(folderPath, { recursive: true });
+    })
 }
 
-const getParserOptions = () => ({
-  metrics: {
-    token: process.env.LogzioMetricsToken,
-    host: process.env.LogzioMetricsHost,
-  },
-  logs: {
-    token: process.env.LogzioLogsToken,
-    host: process.env.LogzioLogsHost,
-  },
-});
-
-module.exports = function processEventHubMessages(context, eventHubMessages, {
-  enableMetric = false,
-}) {
-  const {
-    host,
-    token,
-  } = enableMetric ? getParserOptions().metrics : getParserOptions().logs;
-  context.log(`Starting Logz.io Azure function with enableMetrics: ${enableMetric}`);
-  context.log(JSON.stringify(eventHubMessages));
-  const callBackFunction = getCallBackFunction(context);
-
-  const logzioShipper = logger.createLogger({
-    token,
-    host,
-    type: 'eventHub',
-    protocol: 'https',
-    internalLogger: context,
-    compress: true,
-    debug: true,
-    callback: callBackFunction,
-  });
-
-  const dataParser = new DataParser({
-    internalLogger: context,
-    enableMetric,
-  });
+const uploadFiles = async (containerClient, filesToUpload, context) => {
+    try {
+        const promises = [];
+        for (const file of filesToUpload) {
+            const blockBlobClient = containerClient.getBlockBlobClient(file);
+            promises.push(blockBlobClient.uploadFile(file));
+        }
+        await Promise.all(promises);
+        if (!arrayIsEmpty(filesToUpload)) {
+            context.log('Uploaded logs to back up container.');
+        }
+    } catch (error) {
+        context.log('Error: ', error);
+    }
+};
+const writeEventToBlob = async (event, filesToUpload, foldersToDelete, context) => {
+    if (logIndex >= maxBatchSize) {
+        fileName = createFileName();
+        updateFolderSize();
+        logIndex = 1;
+    } else {
+        logIndex++;
+        const eventWithNewLine = JSON.stringify(event).concat('\n');
+        const concatFolderToFile = folderName + "\\" + fileName;
+        const filePath = workingDir + "\\" + concatFolderToFile;
+        try {
+            await appendFileAsync(filePath, eventWithNewLine);
+            pushFileToUploadAsync(filesToUpload, concatFolderToFile, context);
```

Comment:
inline those methods, the names doesn't make sense and they're really small and self explanatory